mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -285,25 +285,28 @@ class MultiBroadcastConverter final
   }
 };
 
-class DotConverter final : public OpConversionPattern<migraphx::DotOp> {
+template <typename DotType>
+class DotConverter final : public OpConversionPattern<DotType> {
 public:
-  using OpConversionPattern<migraphx::DotOp>::OpConversionPattern;
+  using OpConversionPattern<DotType>::OpConversionPattern;
 
   LogicalResult
-  matchAndRewrite(migraphx::DotOp op, OpAdaptor adaptor,
+  matchAndRewrite(DotType op,
+                  typename OpConversionPattern<DotType>::OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const final {
     Location loc = op->getLoc();
     TypedValue<TensorType> in_A = op.getInA();
     TypedValue<TensorType> in_B = op.getInB();
     auto results = op->getResults();
     auto elementTy =
-        op->getOperand(0).getType().cast<ShapedType>().getElementType();
-    ShapedType outputTy = results[0].getType().cast<ShapedType>();
+        in_A.getType().template cast<ShapedType>().getElementType();
```

Comment:
This shouldn't need the template cast<ShapedType> now as it is a TypedValue< TensorType >. i.e. getElementType(); should work out of the box.

Reply:
Good point! I didn't realize that this is already implicitly casted to a `TensorType` (instead of just a generic `Type`) and who has a method `getElementType()`